src/Shared/FileUtilities.cs

```diff
@@ -266,12 +268,26 @@ internal static bool IsSlash(char c)
         /// <summary>
         /// Trims the string and removes any double quotes around it.
         /// </summary>
-        internal static string TrimAndStripAnyQuotes(string path)
+        ///
+        public static bool EnsurePathIsSingleQuoted(string path)
         {
-            // Trim returns the same string if trimming isn't needed
-            path = path.Trim();
-            path = path.Trim(new char[] { '"' });
+            int endId = path.Length - 1;
+            char singleQuote = '\'';
 
+            return path[0] == singleQuote && path[endId] == singleQuote;
+        }
+        internal static string TrimAndStripAnyQuotes(string path)
+        {
+            
+            if (EnsurePathIsSingleQuoted(path))
```

Comment:
TrimAndStripAnyQuotes explicitly removes double quotes but not single quotes, although that's a little confusing from the name. Going with your version for a moment, this would fail if there are single quotes on one side but not the other or if there is a mix of single and double quotes on the ends.